### ISSUE TITLE
Include 3rd-party dependencies in repository

### ISSUE THREAD

**Issue by briandealwis** [GoogleCloudPlatform/gcloud-eclipse-tools#94]
Installing into a fresh IDE fails due to a missing dependency on `com.google.gson`:

```
Cannot complete the install because one or more required items could not be found.
  Software being installed: Google Cloud Platform for Eclipse 4.5 0.1.0.201604220204 (com.google.cloud.tools.eclipse.suite.e45.feature.feature.group 0.1.0.201604220204)
  Missing requirement: GCloud for Eclipse App Engine Local Server 0.1.0.201604220204 (com.google.cloud.tools.eclipse.appengine.localserver 0.1.0.201604220204) requires 'bundle com.google.gson 2.2.4' but it could not be found
  Cannot satisfy dependency:
    From: Google Cloud Platform for Eclipse 4.5 0.1.0.201604220204 (com.google.cloud.tools.eclipse.suite.e45.feature.feature.group 0.1.0.201604220204)
    To: com.google.cloud.tools.eclipse.appengine.localserver [0.1.0.201604220204]
```

**Comment by elharo** [comment]
http://mvnrepository.com/artifact/com.google.code.gson/gson

**Comment by briandealwis** [comment]
p2 doesn't check Maven Central.  I suppose we could add a reference to Orbit, but I'll just gather up these 3rd party bundles into a new feature.

**Comment by elharo** [comment]
I'm trying to understand where the jars are coming from. We're getting everything from orbit? So what do we do if we want to use a library that's not in orbit? It looks like the version of guava they have is several major versions behind, though that probably doesn't matter for us yet.

**Comment by briandealwis** [comment]
Yeah, we're pulling dependencies from Orbit as it simplifies our target platform setup. PDE/p2 doesn't know how to pull references from Maven Central. And some Maven artifacts aren't bundles (dunno about c.g.gson).

Providing c.g.gson is a bundle, I wonder if m2e and Tycho might work in a hybrid mode here. I'll experiment.

**Comment by briandealwis** [comment]
Update: we can pull in artifacts directly from Maven Central providing they have OSGi metadata (like `com.google.gson`). But it requires some hackery to get them into an Eclipse TP, which doesn't know how to fetch from MC.

**Comment by briandealwis** [comment]
Pointed to the [`p2-maven-plugin`](http://projects.reficio.org/p2-maven-plugin/1.1.0/manual.html) which may help address the TP problem.

**Comment by briandealwis** [comment]
Fixed with #220, #235, and #238
